jxl/src/util/test.rs

```diff
@@ -67,6 +79,17 @@ macro_rules! assert_almost_eq {
             _ => {}
         }
     };
+    ($left:expr, $right:expr, $max_error_abs:expr, $max_error_rel:expr $(,)?) => {
+        let (left_val, right_val, max_error_abs, max_error_rel) = (&$left, &$right, &$max_error_abs, &$max_error_rel);
```

Comment:
Why take a reference?
Also, I feel that there should be more concise ways to express `into` and/or `min` -- perhaps we could just decide to cast everything to f64 and call it a day, as I think this should be used just with f32 and f64